kingpin/actors/packagecloud.py

```diff
@@ -0,0 +1,400 @@
+
+                    # Decrement list of packages to track how many are left
+                    number_in_repo = number_in_repo - 1
+                    # Keeping track of removed packages for return value
+                    deleted_packages.append(package)
+
+        raise gen.Return(deleted_packages)
+
+
+class Delete(PackagecloudBase):
+
+    """packagecloud Delete Actor Class
+
+    **Options**
+
+    :number_to_keep:
+      Keep at least this number of each package
+      (defaults to *0*)
+
+    :packages_to_delete:
+      Regex of packages to delete, e.g. pkg1|pkg2
+
+    :repo:
+      Which packagecloud repo to delete from
+
+    **Examples**
+
+    .. code-block:: json
+
+    { "desc": "packagecloud Delete example",
+      "actor": "packagecloud.Delete",
+      "options": {
+          "number_to_keep": 10,
+          "packages_to_delete": "deleteme",
+          "repo": "test"
+      }
+    }
+
+    """
+
+    all_options = {
+        'number_to_keep': (
+            int, 0,
+            'Keep at least this number of each package'),
+        'packages_to_delete': (
+            str, REQUIRED,
+            'Regex of packages to delete, e.g. pkg1|pkg2'),
+        'repo': (
+            str, REQUIRED,
+            'Which packagecloud repo to delete from'),
+    }
+
+    @gen.coroutine
+    def _execute(self):
+        """Deletes all packages that match the `packages_to_delete` pattern"""
+        yield self._delete(
+            packages_to_delete=self.option('packages_to_delete'),
+            number_to_keep=self.option('number_to_keep'),
+            repo=self.option('repo'))
+
+
+class DeleteByDate(PackagecloudBase):
+
+    """packagecloud DeleteByDate Actor Class
+
+    **Options**
+
+    :number_to_keep:
+      Keep at least this number of each package
+      (defaults to *0*)
+
+    :older_than:
+      Delete packages created before this number of seconds
+
+    :packages_to_delete:
+      Regex of packages to delete, e.g. pkg1|pkg2
+
+    :repo:
+      Which packagecloud repo to delete from
+
+    **Examples**
+
+    .. code-block:: json
+
+    { "desc": "packagecloud DeleteByDate example",
+      "actor": "packagecloud.DeleteByDate",
+      "options": {
+          "number_to_keep": 10,
+          "older_than": 600,
+          "packages_to_delete": "deleteme",
+          "repo": "test"
+      }
+    }
+
+    """
+
+    all_options = {
+        'number_to_keep': (
+            int, 0,
+            'Keep at least this number of each package'),
+        'older_than': (
+            int, REQUIRED,
+            'Delete packages created before this number of seconds'),
+        'packages_to_delete': (
+            str, REQUIRED,
+            'Regex of packages to delete, e.g. pkg1|pkg2'),
+        'repo': (str, REQUIRED,
+                 'Which packagecloud repo to delete from')
+    }
+
+    @gen.coroutine
+    def _execute(self):
+        """Deletes all packages that match the `packages_to_delete` pattern and
```

Comment:
Move this description to the Class. I don't think the _execute method needs a description. It's always "do the actual work for this class"